### ISSUE TITLE
`color-contrast` fails on `aria-hidden="true"` elements

### ISSUE THREAD

**Issue by NickAb** [dequelabs/axe-core#3659]
### Product

axe-core

### Product Version

4.4.3

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation

Elements marked `aria-hidden="true"` should be skipped by `color-contrast` rule as purely decorative (https://www.w3.org/TR/WCAG22/#contrast-minimum).

#### Actual
`color-contrast` rule fails on elements with low contrast intended to be decorative and marked `aria-hidden="true"`.

#### How to Reproduce

html snippet

```html
<i class="material-icon-i material-icons-extended _ngcontent-agc-196" role="img" aria-hidden="true">brightness_1</i>
```

sample unit test

```js
// test/rule-matches/color-contrast-matches.js

it('should not match aria-hidden=true', function () {
    fixture.innerHTML = '<i aria-hidden="true">hi</i>';
    var target = fixture.querySelector('i');
    axe.testUtils.flatTreeSetup(fixture);
    assert.isFalse(rule.matches(target, axe.utils.getNodeFromTree(target)));
});
```

#### Additional context

https://www.w3.org/TR/WCAG22/#contrast-minimum

> The visual presentation of text and images of text has a contrast ratio of at least 4.5:1, except for the following:
> ...
>
>- Incidental: Text or images of text that are part of an inactive user interface component, that are **pure decoration**, that are not visible to anyone, or that are part of a picture that contains significant other visual content, have no contrast requirement.
> ...

**Comment by NickAb** [comment]
Please let me know if `aria-hidden="true"` is not the correct way to mark purely decorative elements and they should be marked differently according tostandards.

**Comment by WilcoFiers** [comment]
@NickAb That looks like a font icon, not just a text correct? Can you clarify if this is reported as a violation or if it is reported as "needs review"? If you're using the browser extension, it is indicated with a little yellow flag icon.

Something like a font icons do have a minimum contrast. It'd 3:1 though. Axe-core can't definitively determine what's going on in such cases so it lets you know they still need to be reviewed.

**Comment by straker** [comment]
The problem is that there is no standard way to mark a visual element as purely decorative (that I know of). `aria-hidden` doesn't make it decorative, it just removes it from the accessibility tree. Visual users may still need the information. Without something to key off of, axe-core can't make any determinations about what color contrast should or should not run against.

**Comment by WilcoFiers** [comment]
I was very much under the impression that axe-core was already ignore icon fonts under color-contrast. It doesn't seem to, so that's going to need to be fixed. I think that's regardless of whether or not aria-hidden is on it. Icon fonts aren't text content, so the rule shouldn't be applied to it. Informative icons have a contrast requirement of 3:1 in WCAG 2.1's non-text contrast.

**Comment by NickAb** [comment]
Here is a codepen - https://codepen.io/nickab/pen/eYrdKeo

As you can see, the icon is purely decorative spacer. As such, I'd expect axe to ignore it. I.e.

1. Apply

```diff
diff --git a/test/integration/rules/color-contrast/color-contrast.html b/test/integration/rules/color-contrast/color-contrast.html
index a1e2ed09..6e306ae8 100644
--- a/test/integration/rules/color-contrast/color-contrast.html
+++ b/test/integration/rules/color-contrast/color-contrast.html
@@ -273,3 +273,11 @@
     felis ante non libero.
   </span>
 </p>
+<p style="color: #dadce0; background-color: #fff">
+  <i
+    id="ignore7"
+    class="material-icon-i material-icons-extended"
+    style="font-family: Google Material Icons"
+    >brightness_1</i
+  >
+</p>
```
2. Run `npm run build && npm run test:unit:integration`
3. - Expect: tests pass
   - Actual: tests fail

But as discussed above, it looks like there is no standard way to mark an element as purely decorative.

**Comment by NickAb** [comment]
btw, purely decorative icons aside, it looks like [lib/checks/color/color-contrast-evaluate.js](https://github.com/dequelabs/axe-core/blob/develop/lib/checks/color/color-contrast-evaluate.js) does not have a special case to handle 3:1 requirements for icons. The required ratio is determined based on text size only.

**Comment by tounsoo** [comment]
Oh.. I've been searching for this all day lol. I guess I have to wait for it but when it does get the update, is `aria-hidden` the correct way to mark pure decorative element?

**Comment by padmavemulapati** [comment]
Validate with the latest axe-core develop branch code base, 
`color-contrast` rule is failing when ligature icon test snippet:
```
<p style="color: #dadce0; background-color: #fff">
     <i
        id="ignore7"
        class="material-icon-i material-icons-extended"
        style="font-family: Google Material Icons"
       >brightness_dull</i
     >
    </p>
```
Passing when 
```
<div id="target" style="font-family: \'Material Icons\'">delete</div>
  <div id="target" style="font-family: \'Material Icons\'">
    check star favorite
  </div>
 <i class="material-icon-i material-icons-extended _ngcontent-agc-196" role="img" aria-hidden="true">brightness</i>

```

![image](https://user-images.githubusercontent.com/47409579/214513857-648fc607-a6e6-4818-b3ba-42e5f1bd54d9.png)
